### PR TITLE
[Skills Everywhere][#144][PORT] WaterfallSkillBotDotNet to JavaScript

### PR DESCRIPTION
**Pull request by denscollo**
Addresses #144

## Description
This PR ports WaterfallSkillBot DotNet to WaterfallSkillBot JS using the following [sample](https://github.com/microsoft/BotFramework-FunctionalTests/tree/main/Bots/DotNet/Skills/CodeFirst/WaterfallSkillBot).

**Known issue:** Restify server url shows `[::]` locally and `undefined:undefined` in Azure causing the samples for Proactive and Cards Audio to not work properly in the functional tests due to the malformed url. (Related issues from Restify repo [`server.url` is wrong](https://github.com/restify/node-restify/issues/1029) and [server.url "undefined" on production server](https://github.com/restify/node-restify/issues/1274).

## Testing
In the following image, there is a test run using WaterfallHostBotJS and WaterfallSkillBotJS.
image

### CODE REVIEW

**Comment by gabog** [comment]
Hi @joshgummersall, can you also help reviewing this one? 
Thanks

File: Bots/JavaScript/Skills/CodeFirst/WaterfallSkillBot/dialogs/delete/deleteDialog.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    /**
+     * @param {import('botbuilder-dialogs').WaterfallStepContext} stepContext
+     */
+    async HandleDeleteDialog(stepContext) {
+        let channel = stepContext.context.activity.channelId;
+
+        if (DeleteDialog.isDeleteSupported(channel))
+        {
+            var id = await stepContext.context.sendActivity(MessageFactory.text("I will delete this message in 5 seconds"));
```
Comment (joshgummersall) [inline Bots/JavaScript/Skills/CodeFirst/WaterfallSkillBot/dialogs/delete/deleteDialog.js:37]: `const`

File: Bots/JavaScript/Skills/CodeFirst/WaterfallSkillBot/dialogs/delete/deleteDialog.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            ]));
+
+        this.initialDialogId = WATERFALL_DIALOG;
+    }
+
+    /**
+     * @param {import('botbuilder-dialogs').WaterfallStepContext} stepContext
+     */
+    async HandleDeleteDialog(stepContext) {
+        let channel = stepContext.context.activity.channelId;
```
Comment (joshgummersall) [inline Bots/JavaScript/Skills/CodeFirst/WaterfallSkillBot/dialogs/delete/deleteDialog.js:33]: `const`, never reassigned.

File: Bots/JavaScript/Skills/CodeFirst/WaterfallSkillBot/dialogs/update/updateDialog.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                var tuple = this.updateTracker[conversationId];
+                var activity = MessageFactory.text(`This message has been updated ${tuple[1]} time(s).`)
+                activity.id = tuple[0];
+                tuple[1]++;
+                this.updateTracker[conversationId] = tuple;
+                await stepContext.context.updateActivity(activity);
+            }
+            else
+            {
+                var id = await stepContext.context.sendActivity(MessageFactory.text("Here is the original activity"));
```
Comment (joshgummersall) [inline Bots/JavaScript/Skills/CodeFirst/WaterfallSkillBot/dialogs/update/updateDialog.js:50]: `const`

File: Bots/JavaScript/Skills/CodeFirst/WaterfallSkillBot/dialogs/activityRouterDialog.js [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+                this.processActivity.bind(this)
+            ]));
+
+        this.initialDialogId = WATERFALL_DIALOG;
+    }
+
+    /**
+     * @param {string} dialogId
+     * @param {import('botbuilder').ConversationState} conversationState
+     * @param {import('../skillConversationIdFactory').SkillConversationIdFactory} conversationIdFactory
```
[the remark this answers is not in the document]
Comment (denscollo) [reply Bots/JavaScript/Skills/CodeFirst/WaterfallSkillBot/dialogs/activityRouterDialog.js:60]: Hi @joshgummersall , this is the doc style is consistent with the rest of the JS bots in this repository, but we can keep it in mind for later.
Thanks!

[the remark this answers is not in the document]
Comment (joshgummersall) [reply Bots/JavaScript/Skills/CodeFirst/WaterfallSkillBot/dialogs/activityRouterDialog.js:60]: Good deal - thanks!

**Review by joshgummersall — APPROVED** [review]

**Review by gabog — APPROVED** [review]

**Pull request merged**